circuits/kimchi/src/polynomials/chacha.rs

```diff
@@ -342,71 +342,81 @@ pub fn chacha20(mut s: Vec<u32>) -> Vec<u32> {
     s
 }
 
-// TODO: Assert somehow that we're using the right number of alphas.
-pub fn constraint<F: FftField>(alpha0: usize) -> E<F> {
-    let w = |i: usize, r: CurrOrNext| E::<F>::cell(Column::Witness(i), r);
+struct Helper<F>(PhantomData<F>);
```

Comment:
converted a lot of the closures to functions. (No need to use closures as we don't capture outside values.)

Reply:
(I think ideally we should return all the constraints and combine them at the end of the main function `constraint`)

Reply:
What is the point of this struct rather than just having functions?

Reply:
oh there's no real benefit besides to avoid having to declare the generic parameters in the functions every time